File: contest/abc328/b.go
package main

import (
	"bufio"
	"fmt"
	"golang.org/x/exp/slices"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	var n int
	fmt.Fscan(in, &n)

	days := make([]int, n)
	for i := 0; i < n; i++ {
		var day int
		fmt.Fscan(in, &day)
		days[i] = day
	}

	count := 0
	for i := 1; i <= n; i++ {
		month := strconv.Itoa(i)
		for d := 1; d <= days[i-1]; d++ {
			day := strconv.Itoa(d)
			tmp := strings.Split(month+day, "")
			uniq := slices.Compact(tmp)
			if len(uniq) == 1 {
				count++
			}
		}
	}

	fmt.Fprintln(out, count)
}

File: contest/abc328/b.rb
n = gets.chomp.to_i
days = gets.chomp.split(" ").map(&:to_i)

count = 0
days.each.with_index(1) do |day, month|
  (1..day).each do|d|
    tmp = month.to_s.split("") + d.to_s.split("").flatten
    count += 1 if tmp.uniq.size == 1
  end
end

puts count